tools/github/setup_docs_env.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+# Options:
+#
+# PIP_FLAGS:
+#   These options are passed to pip.
+
+set -ex
+
+export PIP_DEFAULT_TIMEOUT=60
+
+echo "Installing docs / examples dependencies..."
+# Use previous installed requirements as well, otherwise installing
+# successively may update previously constraint dependencies
+python -m pip install $PIP_FLAGS $REQUIREMENT_FILES -r ./requirements/docs.txt
```

Comment:
Should we make sure that `$REQUIREMENT_FILES` includes `default.txt`, and `optional.txt` (if needed) regardless of `setup_test_env.sh` being run? Also are we sure that `$REQUIREMENT_FILES` sticks from running `setup_test_env.sh` before?

Reply:
This is a remnant from the previous script; I will remove $REQUIREMENT_FILES, since there's really no need to robustly check the docs build under minimum AND latest dependencies. User should only need to be able to build docs after `pip install -r requirements/docs.txt`.

Docs are a small subset of functionality that should be covered by regular test suite.
Docs examples should work, of course, which is why it needs to be run once. But library functionality should be tested in the test suite, and all platforms.

Reply:
Hmm, might be good to check that there's a working dependency solution between "default", "test", "optional" and "docs" for local development. But agreed not critical right now. Might be something we think about, when / if we update our dependency groups though.

Reply:
We could build some checking logic into `tools/generate_requirements.py`?